cmd/kubeadm/test/cmd/version_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeadm
+
+import (
+	"regexp"
+	"testing"
+)
+
+const (
+	ShortExpectedRegex  = "^v.+\n$"
+	JsonExpectedRegex   = "^{\"kubeadmVersion\":{\"major\":\".+\",\"minor\":\".+\",\"gitVersion\":\".+\",\"gitCommit\":\".+\",\"gitTreeState\":\".+\",\"buildDate\":\".+\",\"goVersion\":\".+\",\"compiler\":\".+\",\"platform\":\".+\"}}\n$"
+	YamlExpectedRegex   = "^kubeadmVersion:\n\\s+buildDate:.+\n\\s+compiler:.+\n\\s+gitCommit:.+\n\\s+gitTreeState:.+\n\\s+gitVersion:.+\n\\s+goVersion:.+\n\\s+major:.+\n\\s+minor:.+\n\\s+platform:.+\n\n$"
```

Comment:
I think instead of having a regex for json and yaml, you could just marshal it and check it that way? The json and yaml regex seems a bit intense.